azure-devops/javascript/stages/50-deployment/k8s.yaml

```diff
@@ -14,6 +14,12 @@ stages:
               dockerRegistryEndpoint: '$(dockerRegistryEndpoint)'
               kubernetesServiceConnection: '$(k8sEndpoint)'
               namespace: '$(k8sNamespace)'
+          - script: |
```

Comment:
Instead of doing that way, could you please just check the YAML files, iterate for all environments you have inside the YAML file, and try to add them to the file? For example:
```
- script: |
    for var in $(compgen -e); do
      sed -i "s/\$$var/${!var}/g" yourfile.yaml
    done
  displayName: 'Replace Environment Variables'
```